From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Stultz <john.stultz@linaro.org>
Date: Fri, 17 Apr 2020 22:43:21 +0000
Subject: ANDROID: iommu: Kconfig: Make ARM_SMMU depend on QCOM_SCM ||
 !QCOM_SCM

The arm-smmu-qcom code calls qcom_scm_qsmmu500_wait_safe_toggle().
Since the QCOM_SCM config can now be set as a module, this
requires the ARM_SMMU also be a module in that case, so make
the dependency explicit in the Kconfig.

This is done via the non-intuitive "depends on QCOM_SCM || !QCOM_SCM"
Kconfig syntax, which basically ensures that QCOM_SCM isn't
configured as a module if ARM_SMMU is built in.

Fixes: 6a1652effdb8 ("ANDROID: firmware: QCOM_SCM: Allow qcom_scm driver to be loadable as a permenent module")
Signed-off-by: John Stultz <john.stultz@linaro.org>
Change-Id: I580e65e3df85bfbda2af5c54a1a068f6de7d27ee
---
 drivers/iommu/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/iommu/Kconfig b/drivers/iommu/Kconfig
--- a/drivers/iommu/Kconfig
+++ b/drivers/iommu/Kconfig
@@ -382,6 +382,7 @@ config SPAPR_TCE_IOMMU
 config ARM_SMMU
 	tristate "ARM Ltd. System MMU (SMMU) Support"
 	depends on (ARM64 || ARM || (COMPILE_TEST && !GENERIC_ATOMIC64)) && MMU
+	depends on QCOM_SCM || !QCOM_SCM #if QCOM_SCM=m this can't be =y
 	select IOMMU_API
 	select IOMMU_IO_PGTABLE_LPAE
 	select ARM_DMA_USE_IOMMU if ARM
